etna/datasets/internal_datasets.py

```diff
@@ -239,6 +254,99 @@ def get_m4_dataset(dataset_dir: Path, dataset_freq: str) -> None:
+    Jan. 1st 2008 to Mar. 30th 2009 (15 days were dropped from this period: public holidays and two days with
+    anomalies, we set zero values for these days). Initial dataset has 10 min frequency, we create traffic with hour
+    frequency by mean aggregation. Each frequency mode has its own specific prediction horizon: 6 * 24 for 10T,
+    24 for hourly.
+
+    Parameters
+    ----------
+    dataset_dir:
+        The path for saving dataset locally.
+    dataset_freq:
+        Frequency mode.
+
+    Notes
+    -----
+    There is another "traffic" dataset that is also popular and used in papers for time series tasks. This
+    dataset is also from the California Department of Transportation PEMS website, http://pems.dot.ca.gov, however for
+    different time period: from 2015 to 2016.
+
+    References
+    ----------
+    .. [1] https://archive.ics.uci.edu/dataset/204/pems+sf
+    .. [2] http://pems.dot.ca.gov
+    """
+
+    def read_data(path: Path, part: str) -> np.ndarray:
+        with open(path, "r") as f:
+            if part in ("randperm", "stations_list"):
+                data = f.read().lstrip("[").rstrip("]\n").split(" ")
+                out = np.array(map(int, data)) if part == "randperm" else np.array(data)
+                return out
+            else:
+                lines = []
+                for line in f:
+                    line_segments = line.lstrip("[").rstrip("]\n").split(";")
+                    line_target = [list(map(float, segment.split(" "))) for segment in line_segments]
+                    lines.append(line_target)
+                out = np.array(lines)
+                return out
+
+    url = "https://archive.ics.uci.edu/static/public/204/pems+sf.zip"
+
+    dataset_dir.mkdir(exist_ok=True, parents=True)
+
+    file_names = ("randperm", "stations_list", "PEMS_train", "PEMS_test")
+    read_functions = tuple(partial(read_data, part=file_name) for file_name in file_names)
+
+    ts_indecies, stations, targets_train, targets_test = _download_dataset_zip(
+        url=url, file_names=file_names, read_functions=read_functions
+    )
+
+    targets = np.concatenate([targets_train, targets_test], axis=0)
+    targets = targets[np.argsort(ts_indecies)].reshape(-1, 963)
+
+    drop_days = (
+        list(holidays.UnitedStates(years=2008).keys())
+        + list(holidays.UnitedStates(years=2009).keys())[:3]
+        + [date(2009, 3, 8), date(2009, 3, 10)]
+    )
+
+    dates_df = pd.DataFrame({"timestamp": pd.date_range("2008-01-01 00:00:00", "2009-03-30 23:50:00", freq="10T")})
+    dates_df["dt"] = dates_df["timestamp"].dt.date
+    dates_df_cropped = dates_df[~dates_df["dt"].isin(drop_days)]
+    dates_df = dates_df.drop(["dt"], axis=1)
+
+    df = pd.DataFrame(targets, columns=stations)
+    df["timestamp"] = dates_df_cropped["timestamp"].values
+    df = df.merge(dates_df, on=["timestamp"], how="right").fillna(0)
```

Comment:
What columns are you merging here?

Reply:
`dates_df` has all the days from 2008-01-01 to 2009-03-30, but `traffic` has dropped days (15 days) which we decide to define as zeros, so I do a `right join` here to add the missing dates to `df` and replace them with 0.